### PR TITLE
Fix the custom image upload

### PR DESCRIPTION
**Pull request by Gauravjeetsingh**
### This fixes issue #1699 

custom-theme

### Also, fixes the issue #1698 

image

### CODE REVIEW

File: www/main/settings/utils/theme-bg-uploader.js [hunk 1]
```diff
@@ -51,29 +52,39 @@ export const uploadImage = async (evt) => {
...
+            }
+          });
+      } else {
+        errorAlert('File must be in .png or .jpg format');
+        reject();
       }
-    } else {
-      errorAlert('File must be in .png or .jpg format');
+    } catch (error) {
+      errorAlert('Unknown error occured');
```
Comment (theamanjs) [inline www/main/settings/utils/theme-bg-uploader.js:86]: I think this should come from i18n json file

File: www/main/settings/utils/theme-bg-uploader.js [hunk 2]
```diff
@@ -51,29 +52,39 @@ export const uploadImage = async (evt) => {
...
-        const customThemeObj = {
-          type: 'custom',
-          url: `${fileUrl}`.replace(/(%20)/g, '\\ '),
-        };
+      if (imageCheck(filePath)) {
+        sharp(filePath)
+          .jpeg({ mozjpeg: true })
+          .toFile(newPath, (err) => {
+            if (err) {
+              errorAlert('Unable to save file');
```
Comment (theamanjs) [inline www/main/settings/utils/theme-bg-uploader.js:67]: This should come from i18n

File: www/main/settings/utils/theme-bg-uploader.js [hunk 3]
```diff
@@ -51,29 +52,39 @@ export const uploadImage = async (evt) => {
...
-        GlobalState.getActions().userSettings.setThemeBg(customThemeObj);
-        global.core.platformMethod('updateSettings');
+              store.setUserPref('app.themebg', customThemeObj);
+              GlobalState.getActions().userSettings.setThemeBg(customThemeObj);
+              global.core.platformMethod('updateSettings');
+              resolve();
+            }
+          });
+      } else {
+        errorAlert('File must be in .png or .jpg format');
```
Comment (theamanjs) [inline www/main/settings/utils/theme-bg-uploader.js:82]: Can this come from i18n?

**Review by theamanjs — CHANGES_REQUESTED** [review]

File: www/main/settings/utils/theme-bg-uploader.js (hunk 1 above)
Comment (Gauravjeetsingh) [reply www/main/settings/utils/theme-bg-uploader.js]: Done.

File: www/main/settings/utils/theme-bg-uploader.js (hunk 2 above)
Comment (Gauravjeetsingh) [reply www/main/settings/utils/theme-bg-uploader.js]: Done.

File: www/main/settings/utils/theme-bg-uploader.js (hunk 3 above)
Comment (Gauravjeetsingh) [reply www/main/settings/utils/theme-bg-uploader.js]: Done.

**Comment by theamanjs** [comment]
LGTM

**Review by theamanjs — DISMISSED** [review]

File: www/main/settings/components/Setting.jsx [hunk 4]
```diff
@@ -31,15 +31,18 @@ const Setting = ({ settingObj, stateVar, stateFunction }) => {
...
-          type="range"
-          data-value={userSettings[stateVar]}
-          value={userSettings[stateVar]}
-          min={min}
-          max={max}
-          step={step}
-          onChange={handleInputChange}
-        ></input>
+        <>
+          <p className='range-value'>{userSettings[stateVar]}</p>
```
Comment (theamanjs) [inline www/main/settings/components/Setting.jsx:35]: @Gauravjeetsingh veerji I'm seeing this what's a new window. This is supposed to be removed.

Screenshot 2023-01-02 at 11 08 18 PM

**Review by theamanjs — CHANGES_REQUESTED** [review]

File: www/main/settings/components/Setting.jsx (hunk 4 above)
Comment (theamanjs) [reply www/main/settings/components/Setting.jsx:35]: https://github.com/KhalisFoundation/sttm-desktop/issues/1701 opened in separate issue. I think we are good to go with this one

**Review by theamanjs — APPROVED** [review]

**Pull request merged**